### DISCUSSION TITLE
Scaffold DBT graph operators into config at run time

### DISCUSSION THREAD

**Discussion by fredonia88** [dagster-io/dagster#20244] [Q&A] [answered]
What is the best way to scaffold dbt graph operators into a select string at run time from a ConfigurableResource? I'm trying to add `+` or `@` to my dbt selection strings when materializing a dbt asset from the Dagster UI -- meaning, I click on an asset and decide I want to run all its parents, so I update the config to include `+` in a class variable. Is there a way I can override the select statement in the `dbt.cli` object with the operator? 

I tried updating the `context.selected_output_names` set with the graph operators and then assigning the result back to `context.selected_output_names` before streaming `dbt.cli`, but there is no setter available.

What is the best way to scaffold graph operators into a dbt select string from within `@dbt_assets`?

**Comment by fredonia88** [comment]
hey @rexledesma just wanted see if you had any thoughts on the how to incorporate dbt graph operators into a dagster config? thank you for your time.

**Comment by benpankow** [comment, marked as answer]
Hi Frederick - could you share a bit more about what your asset structure looks like right now? Typically, if the parent dbt models are are represented in Dagster, selecting them in the asset graph before clicking materialize should be sufficient to get dbt to run all of them - the asset/output selection passed in the `context` to `dbt.cli(["build"], context=context)` is converted to the corresponding dbt selectors (`+`, `@` etc)

**Reply by fredonia88** [reply]
Hey @benpankow - My asset structure mostly resembles a typical dbt project layout: staging -> intermediate -> marts. I did not realize how easy it was to select all parents and children directly within the graph! Thanks for pointing that out -- I'll mark your response as answer.
